### PR TITLE
Exception handling in case of hyperparameters tuning after fitting

### PR DESCRIPTION
**Pull request by rednevaler**
Before submitting a pull request, please do the following steps:

1. Read [instructions for contributors](https://catboost.ai/docs/concepts/development-and-contributions.html).
2. Run `ya make` in catboost folder to make sure the code builds.
3. Add tests that test your change.
4. Run tests using `ya make -t -A` command.
5. If you haven't already, complete the [CLA](https://yandex.ru/legal/cla/).

### CODE REVIEW

File: catboost/python-package/catboost/core.py [hunk 1]
```diff
@@ -3049,7 +3049,11 @@ def _tune_hyperparams(self, param_grid, X, y=None, cv=3, n_iter=10, partition_ra
                 search_by_train_test_split, calc_cv_statistics, custom_folds, verbose
             )
 
-        self.set_params(**cv_result['params'])
+        if not self.is_fitted():
+            self.set_params(**cv_result['params'])
+        else:
+            raise CatBoostError("Model was fitted before hyperparameters tuning. You can't change hyperparameters of fitted model.")
```
Comment (nikitxskv) [inline catboost/python-package/catboost/core.py]: We don't need to set params if we not to refit our model. So let's do like this:
```
if refit:
    if not self.is_fitted():
        raise CatBoostError("Model was fitted before hyperparameters tuning. You can't change hyperparameters of fitted model.")
    self.set_params(**cv_result['params'])
    self.fit(X, y, silent=True)
```

**Comment by nikitxskv** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
CLA already signed

**Comment by arcadia-devtools** [comment]
@nikitxskv, internal review request created: [1176351](https://a.yandex-team.ru/review/1176351)

**Comment by arcadia-devtools** [comment]
Check [1176351](https://a.yandex-team.ru/review/1176351) passed

**Pull request merged**